modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/WFSFeatureSource.java

```diff
@@ -254,6 +254,10 @@ protected GetFeatureRequest createGetFeature(Query query, ResultType resultType)
         if (Integer.MAX_VALUE > maxFeatures && canLimit(query)) {
             request.setMaxFeatures(maxFeatures);
         }
+        Integer startIndex = query.getStartIndex();
+        if (startIndex != null && Integer.MAX_VALUE > startIndex && canOffset(query)) {
```

Comment:
``maxFeatures`` defaults to MAX_VALUE and needs to be tested, but startIndex does not need this check AFAIK.

Reply:
Fixed. Thanks.